File: service-core/src/main/java/com/timothy/service_core/ServiceDatabase.java
package com.timothy.service_core;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import com.timothy.spotifyarchitecture.entities.SpotifyUser;
import com.timothy.spotifyarchitecture.entities.Token;
import com.timothy.spotifyarchitecture.room.CustomTypeConverter;
import com.timothy.spotifyarchitecture.room.SpotifyDao;

/**
 * Created by tim on 5/30/17.
 */
@Database(entities = {Token.class, SpotifyUser.class}, version = 2)
@TypeConverters({CustomTypeConverter.class})
public abstract class ServiceDatabase extends RoomDatabase {
    public abstract SpotifyDao serviceDao();
}

File: app/src/main/java/com/timmahh/musicall/ServiceDatabase.kt
package com.timmahh.musicall

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters
import com.timmahh.spotifyweb.entities.SpotifyAlbum
import com.timmahh.spotifyweb.entities.SpotifyArtist
import com.timmahh.spotifyweb.entities.SpotifyTrack
import com.timmahh.spotifyweb.entities.SpotifyUser
import com.timmahh.spotifyweb.room.SpotifyConverters
import com.timmahh.spotifyweb.room.SpotifyDao
import com.timmahh.spotifyweb.room.SpotifyDatabase

/**
 * Room database class
 */
@Database(entities = arrayOf(SpotifyUser::class, SpotifyAlbum::class,
		SpotifyArtist::class, SpotifyTrack::class), version = 3)
@TypeConverters(SpotifyConverters::class)
abstract class ServiceDatabase : RoomDatabase(), SpotifyDatabase {
	override abstract fun spotifyDao(): SpotifyDao
}